full/src/test/java/apoc/load/LoadGoogleCloudStorageTest.java

```diff
@@ -33,37 +35,30 @@
 
 public class LoadGoogleCloudStorageTest {
 
-    private static final String BUCKET_NAME = System.getenv("APOC_GC_BUCKET_NAME");
-
-    @BeforeClass
-    public static void before() {
-        Assume.assumeNotNull(BUCKET_NAME);
-    }
-
+    public static GoogleCloudStorageContainerExtension gcs = new GoogleCloudStorageContainerExtension()
+            .withMountedResourceFile("test.csv", "/folder/test.csv")
+            .withMountedResourceFile("map.json", "/folder/map.json");
 
-    @Rule
-    public DbmsRule db = new ImpermanentDbmsRule();
+    @ClassRule
+    public static DbmsRule db = new ImpermanentDbmsRule();
 
-    @Before
-    public void setUp() throws Exception {
+    @BeforeClass
+    public static void setUp() throws Exception {
+        gcs.start();
         TestUtil.registerProcedure(db, LoadCsv.class, LoadJson.class);
     }
 
-    @Test
-    public void testLoadCsvWithoutAuth() throws Exception {
-        String url = "gs://" + BUCKET_NAME + "/test.csv";
-        testResult(db, "CALL apoc.load.csv({url})", map("url", url), (r) -> {
-            assertRow(r, "Selma", "8", 0L);
-            assertRow(r, "Rana", "11", 1L);
-            assertRow(r, "Selina", "18", 2L);
-            assertFalse("It should be the last record", r.hasNext());
-        });
+    @AfterClass
+    public static void tearDown() {
+        gcs.close();
+        db.shutdown();
     }
 
     @Test
-    public void testLoadCsvFromFolderWithoutAuth() throws Exception {
```

Comment:
Are the new tests with auth?

Reply:
I don't think so. Again this is a cherry-pick of https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/dev/extended/src/test/java/apoc/load/LoadGoogleCloudStorageTest.java.

The underlying container https://github.com/fsouza/fake-gcs-server#running-with-http doesn't have any option to be passing authentication and test that part of the functionality unfortunately.

Reply:
Please add the cherry picked PR in the PR description here, so we know part of it is cherry picked. Future us might thank us for it.